api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentService.java

```diff
@@ -34,6 +40,35 @@
     private EventAuditor eventAuditor;
     private PaymentCalculator paymentCalculator;
 
+    /**
+     * Method used to store a failed payment to the database with the status of FAILURE. Will attempt to
+     * retrieve the payment amount and reference from the failure event if available.
+     *
+     * @param paymentCycle  The payment cycle for the payment
+     * @param cardAccountId The card account id
+     * @param failureEvent  The failure event.
+     */
+    public void saveFailedPayment(PaymentCycle paymentCycle, String cardAccountId, FailureEvent failureEvent) {
+        try {
+            Map<String, Object> eventMetadata = failureEvent.getEventMetadata();
+            Integer amountToPayInPence = (Integer) eventMetadata.get(PAYMENT_AMOUNT.getKey());
+            String paymentReference = (String) eventMetadata.get(PAYMENT_REFERENCE.getKey());
+            Payment failedPayment = Payment.builder()
+                    .cardAccountId(cardAccountId)
+                    .claim(paymentCycle.getClaim())
+                    .paymentAmountInPence(amountToPayInPence)
+                    .paymentCycle(paymentCycle)
+                    .paymentReference(paymentReference)
+                    .paymentStatus(PaymentStatus.FAILURE)
+                    .build();
+
+            paymentRepository.save(failedPayment);
+        } catch (Exception e) {
+            log.error("Unexpected exception caught saving a failed payment for paymentCycle: {}, cardAccountId: {}, failureEvent: {}, exception detail: {}",
```

Comment:
I didn't see what else we could do should this attempt to save the failed payment also fail.